src/account/AccountUtils.ts

```diff
@@ -0,0 +1,216 @@
+            const jwtClaims = getIssAudAndUidVal(keylessComponents);
+            return new KeylessAccount({ ...keylessComponents, ...jwtClaims });
+          }
+          case AnyPublicKeyVariant.FederatedKeyless: {
+            const keylessComponents = this.deserializeKeylessAccountCommon(deserializer);
+            const jwkAddress = AccountAddress.deserialize(deserializer);
+            const audless = deserializer.deserializeBool();
+            const jwtClaims = getIssAudAndUidVal(keylessComponents);
+            return new FederatedKeylessAccount({ ...keylessComponents, ...jwtClaims, jwkAddress, audless });
+          }
+          default:
+            throw new Error(`Unsupported public key variant ${variantIndex}`);
+        }
+      }
+      case SigningScheme.MultiKey: {
+        const multiKey = MultiKey.deserialize(deserializer);
+        const length = deserializer.deserializeUleb128AsU32();
+        const signers = new Array<SingleKeySignerOrLegacyEd25519Account>();
+        for (let i = 0; i < length; i += 1) {
+          const signer = this.deserialize(deserializer);
+          if (!isSingleKeySigner(signer) && !(signer instanceof Ed25519Account)) {
+            throw new Error(
+              "Deserialization of MultiKeyAccount failed. Signer is not a SingleKeySigner or Ed25519Account",
+            );
+          }
+          signers.push(signer);
+        }
+        return new MultiKeyAccount({ multiKey, signers, address });
+      }
+      default:
+        throw new Error(`Deserialization of Account failed: invalid signingScheme value ${signingScheme}`);
+    }
+  }
+
+  static keylessAccountFromHex(hex: HexInput): KeylessAccount {
+    const account = this.fromHex(hex);
+    if (!(account instanceof KeylessAccount)) {
+      throw new Error("Deserialization of KeylessAccount failed");
+    }
+    return account;
+  }
+
+  static federatedKeylessAccountFromHex(hex: HexInput): FederatedKeylessAccount {
+    const account = this.fromHex(hex);
+    if (!(account instanceof FederatedKeylessAccount)) {
+      throw new Error("Deserialization of FederatedKeylessAccount failed");
+    }
+    return account;
+  }
+
+  static multiKeyAccountFromHex(hex: HexInput): MultiKeyAccount {
+    const account = this.fromHex(hex);
+    if (!(account instanceof MultiKeyAccount)) {
+      throw new Error("Deserialization of MultiKeyAccount failed");
+    }
+    return account;
+  }
+
+  static singleKeyAccountFromHex(hex: HexInput): SingleKeyAccount {
+    const account = this.fromHex(hex);
+    if (!(account instanceof SingleKeyAccount)) {
+      throw new Error("Deserialization of SingleKeyAccount failed");
+    }
+    return account;
+  }
+
+  static ed25519AccountFromHex(hex: HexInput): Ed25519Account {
+    const account = this.fromHex(hex);
+    if (!(account instanceof Ed25519Account)) {
+      throw new Error("Deserialization of Ed25519Account failed");
+    }
+    return account;
+  }
```

Comment:
Do we need these?
It's a very simple check that users can do on their app if needed.
They can also just do the following if they're handling a single type.
```typescript
const ed25519Account = AccountUtils.fromHex(hex) as Ed25519Account;
```

Reply:
I could remove, but its just for convenience and for type safety.  Currently there is no Ed25519Account.deserialize so I'm providing this instead.